opensearch_py_ml/ml_models/sentencetransformermodel.py

```diff
@@ -1006,6 +1008,55 @@ def set_up_accelerate_config(
                 "Failed to open config file for ml common upload: " + file_path + "\n"
             )
 
+    def _get_model_description_from_md_file(self, readme_file_path) -> str:
+        """
+        Get description of the model from README.md file
+        after the model is saved in local directory
+
+        This function assumes that the md file has the following format in the README.md:
+
+        # sentence-transformers/msmarco-distilbert-base-tas-b
+        This is [ ... further description ... ]
+
+        # [ ... Next section ...]
+        ...
+
+        :param readme_file_path: Path to README.md file
+        :type readme_file_path: string
+        :return: Description of the model
+        :rtype: string
+        """
+        readme_data = MarkDownFile.read_file(readme_file_path)
+
+        # Find the description section
+        start_str = f"# {self.model_id}"
+        start = readme_data.find(start_str)
+        if start == -1:
+            model_name = self.model_id.split("/")[1]
+            start_str = f"# {model_name}"
+            start = readme_data.find(start_str)
+        end = readme_data.find("\n#", start + len(start_str))
+
+        # If we cannot find the scope of description section, raise error.
+        if start == -1 or end == -1:
+            assert False, "Cannot find description in README.md file"
+
+        # Parse out the description section
+        description = readme_data[start + len(start_str) + 1 : end].strip()
+        description = description.split("\n")[0]
+
+        # Remove hyperlink and reformat text
+        description = re.sub(r"\(.*?\)", "", description)
+        description = re.sub(r"[\[\]]", "", description)
+        description = re.sub(r"\*", "", description)
+
+        # Remove unnecessary part if exists(e.g. " For an introduction to ..."
+        unnecessary_part = description.find(" For an introduction to")
```

Comment:
I'm not sure how general this description method. Is the readme file always follow such pattern with " For an introduction to" ?

Reply:
If the readme pattern is unknown, or it's hard to predict the pattern. We'd better make the method flexible or more general for future pattern

Reply:
There are couple models that have this "For an introduction to ... " line (https://huggingface.co/sentence-transformers/multi-qa-mpnet-base-dot-v1), so I remove it if it exists since Dhrubo manually removes it from the current model listing. It will do nothing if it doesn't exist though.